mig-blue-green/setup.sh

```diff
@@ -0,0 +1,226 @@
+        echo "Waiting few seconds to retry" &&
+        sleep 10 && \
+        echo "Retrying $COMMAND" && \
+        eval $COMMAND \
+    )
+}
+
+echo "> Enabling required APIs"
+# Some of these can be enabled later with Terraform, but I personally
+# prefer to do all API enablement in one place with gcloud.
+gcloud services enable \
+    --project=$PROJECT_ID \
+    cloudbuild.googleapis.com \
+    cloudresourcemanager.googleapis.com \
+    compute.googleapis.com \
+    sourcerepo.googleapis.com \
+    --no-user-output-enabled \
+    --quiet
+
+echo "> Adding Cloud Build to roles/editor"
+gcloud projects add-iam-policy-binding \
+    "$PROJECT_ID" \
+    --member="serviceAccount:$PROJECT_NUM@cloudbuild.gserviceaccount.com" \
+    --role='roles/editor' \
+    --condition=None \
+    --no-user-output-enabled \
+    --quiet
+
+echo "> Adding Cloud Build to roles/source.admin"
+gcloud projects add-iam-policy-binding \
+    "$PROJECT_ID" \
+    --member="serviceAccount:$PROJECT_NUM@cloudbuild.gserviceaccount.com" \
+    --condition=None \
+    --role='roles/source.admin' \
+    --no-user-output-enabled \
+    --quiet
+
+echo "> Configuring bootstrap job"
+rm -rf "./bootstrap.cloudbuild.yaml"
+cat <<'EOT_BOOT' > "./bootstrap.cloudbuild.yaml"
+tags:
+- "mig-blue-green-bootstrapping"
+steps:
+- id: create_new_cloud_source_repo
+  name: "gcr.io/cloud-builders/gcloud"
+  script: |
+    #!/bin/bash
+    set -e
+
+    echo "(Re)Creating source code repository"
+
+    gcloud source repos delete \
+        "copy-of-mig-blue-green" \
+        --quiet || true
+    
+    gcloud source repos create \
+        "copy-of-mig-blue-green" \
+        --quiet
+
+- id: copy_demo_source_into_new_cloud_source_repo
+  name: "gcr.io/cloud-builders/gcloud"
+  env:
+    - "PROJECT_ID=$PROJECT_ID"
+    - "PROJECT_NUMBER=$PROJECT_NUMBER"
+  script: |
+    #!/bin/bash
+    set -e
+
+    readonly GIT_REPO="https://github.com/GoogleCloudPlatform/cloud-build-samples.git"
+
+    echo "Cloning demo source repo"
+    mkdir /workspace/from/
+    cd /workspace/from/
+    git clone $GIT_REPO
+    cd ./cloud-build-samples
+
+    echo "Cloning new empty repo"
+    mkdir /workspace/to/
+    cd /workspace/to/
+    gcloud source repos clone \
+        "copy-of-mig-blue-green"
+    cd ./copy-of-mig-blue-green
+
+    echo "Making a copy"
+    cp -r /workspace/from/cloud-build-samples/mig-blue-green/* ./
+
+    echo "Setting git identity"
+    git config user.email \
+        "$PROJECT_NUMBER@cloudbuild.gserviceaccount.com"
+    git config user.name \
+        "Cloud Build"
+
+    echo "Commit & push"
+    git add .
+    git commit \
+        -m "A copy of $GIT_REPO"
+    git push
+
+- id: add_pipeline_triggers
+  name: "gcr.io/cloud-builders/gcloud"
+  env:
+    - "PROJECT_ID=$PROJECT_ID"
+  script: |
+    #!/bin/bash
+    set -e
+
+    echo "(Re)Creating destroy trigger"
+    gcloud beta builds triggers delete "destroy" --quiet || true
+    gcloud beta builds triggers create manual \
```

Comment:
Does this need beta?

Reply:
it works fine as it is; I am not sure why it ould need "beta"

Reply:
Please remove beta from all these commands if not needed

Reply:
I removed 'beta' from all but one place. I kept it in the setup.sh.

Reason being that only 'beta' supports streaming of cloud build execution logs to the console. While it is not important for operations in the unattaneded testing, setup.sh is also executed by user directly. There, streaming logs will help the user to see the root cause of possible errors. Without 'beta' a user must find exactly which build failed in one of two different GCP projects, which maybe non trivial.